src/libraries/System.Private.CoreLib/src/System/IO/File.netcoreapp.cs

```diff
@@ -97,5 +103,129 @@ private static byte[] ReadAllBytesUnknownLength(FileStream fs)
                 }
             }
         }
+
+        private static void WriteToFile(string path, FileMode mode, string? contents, Encoding encoding)
+        {
+            ReadOnlySpan<byte> preamble = encoding.GetPreamble();
+            int preambleSize = preamble.Length;
+
+            using SafeFileHandle fileHandle = OpenHandle(path, mode, FileAccess.Write, FileShare.Read, FileOptions.None, GetPreallocationSize(mode, contents, encoding, preambleSize));
+            long fileOffset = mode == FileMode.Append && fileHandle.CanSeek ? RandomAccess.GetLength(fileHandle) : 0;
+
+            if (string.IsNullOrEmpty(contents))
+            {
+                if (preambleSize > 0) // even if the content is empty, we want to store the preamble
```

Comment:
Should this (and others) include a check for `fileOffset == 0`? The `StreamWriter` does not write the preamble when position > 0.

https://github.com/dotnet/runtime/blob/d70fa8fb10327d5142ea5b0b9b980aaeac759fe4/src/libraries/System.Private.CoreLib/src/System/IO/StreamWriter.cs#L127-L132

Reply:
@campersau great catch! I am going to add a test for that and fix it